src/modules/bootloader/main.py

```diff
@@ -605,17 +605,71 @@ def run_grub_install(fw_type, partitions, efi_directory):
         assert efi_directory is not None
         efi_bootloader_id = efi_label(efi_directory)
         efi_target, efi_grub_file, efi_boot_file = get_grub_efi_parameters()
+        grub_modules = get_modules()
+        install_path = libcalamares.globalstorage.value("rootMountPoint")
 
         if is_zfs:
             check_target_env_call(["sh", "-c", "ZPOOL_VDEV_NAME_PATH=1 " + libcalamares.job.configuration["grubInstall"]
                                    + " --target=" + efi_target + " --efi-directory=" + efi_directory
+                                   + " --modules=\"" + grub_modules + "\" --sbat /usr/share/grub/sbat.csv"
+                                   + " --disable-shim-lock"
                                    + " --bootloader-id=" + efi_bootloader_id + " --force"])
         else:
-            check_target_env_call([libcalamares.job.configuration["grubInstall"],
+            check_target_env_call(["sh", "-c", " ".join([libcalamares.job.configuration["grubInstall"],
                                    "--target=" + efi_target,
                                    "--efi-directory=" + efi_directory,
+                                   '--modules="' + grub_modules + '"',
+                                   "--disable-shim-lock",
+                                   "--sbat /usr/share/grub/sbat.csv",
                                    "--bootloader-id=" + efi_bootloader_id,
-                                   "--force"])
+                                   "--force"])])
+        MOK_path = "/etc/secureboot/"
+        if os.path.exists(install_path + MOK_path + "MOK.key"):
+            grub_path = efi_directory + "/EFI/" + efi_bootloader_id + "/" + efi_grub_file
+            check_target_env_call(["sh", "-c", "/usr/bin/find /boot/ -maxdepth 1 -name 'vmlinuz-*' | /usr/bin/xargs -I{} /bin/sh -c 'if ! /usr/bin/sbverify --cert /etc/secureboot/MOK.crt {} >/dev/null 2>&1 ;then sudo /usr/bin/sbsign --key " + MOK_path + "MOK.key --cert " + MOK_path + "MOK.crt --output {} {}; fi'"])
+            check_target_env_call(["sh", "-c", "/usr/bin/sudo /usr/bin/sbsign --key " + MOK_path + "MOK.key --cert " + MOK_path + "MOK.crt --output " + grub_path + " " + grub_path])
+            check_target_env_call(["sh", "-c", "/usr/bin/sudo /usr/bin/mokutil --import " + MOK_path + "MOK.cer --root-pw"])
+            install_efi_directory = install_path + efi_directory
+            shutil.copy2(os.path.join(install_path, "/usr/share/shim-signed/shimx64.efi") , os.path.join(install_efi_directory, "EFI", efi_bootloader_id))
+            shutil.copy2(os.path.join(install_path, "/usr/share/shim-signed/mmx64.efi") , os.path.join(install_efi_directory, "EFI", efi_bootloader_id))
+            shutil.copy2(os.path.join(install_path, "/usr/share/shim-signed/shimx64.efi") , os.path.join(install_efi_directory, "EFI", efi_bootloader_id, "bootx64.efi"))
```

Comment:
Have you tested this?  I think `os.path.join` will discard `install_path` in this situation and so those files are being copied from the host instead of the target.

That may be problematic if the host and target are different.